### PR TITLE
Add details to README (#339)

### PR DESCRIPTION
**Pull request by natalieesk**
Adds content to README
Creates Makefile commands

This closes [#339](https://github.com/ZcashFoundation/frost/issues/339)

### CODE REVIEW

File: README.md
Comment (mpguerra) [inline README.md:20]: It would be good to have details of how to run the demo exactly. Assume no prior knowledge of rust or anything else :)

Comment (mpguerra) [reply README.md:20]: e.g to run this demo, you need to 1. clone this repo, 2. build the demo and 3. run the demo.

Including the commands to do so for each step. This may go in a different separate section. 

I would also add some screenshots of the expected output.

File: README.md
Comment (mpguerra) [inline README.md:31]: Maybe we want to make some comment about how in a "real world" scenario, the secret share should be delivered securely to each participant and not simply printed out to terminal. Additionally, or instead, we might want to explain what each of these items which are output is used for in the next round; which should be kept secret and which should be kept private and how they would ideally be delivered to participants.

I know this is all in the spec but people are lazy about reading links :D

File: README.md
Comment (mpguerra) [inline README.md:41]: I don't understand what this is for :) Why would someone interested in the demo want to execute this command.

File: README.md
Comment (mpguerra) [inline README.md:45]: Same comment here as above. Are these details intended for internal developers?

File: README.md
Comment (mpguerra) [inline README.md:33]: _Optional_: I keep going back and forth about whether this section should be right at the end or closer to the top.

**Review by mpguerra — CHANGES_REQUESTED** [review]
Looks good! I would just add some more details about exactly how to build and run the demo.

File: README.md
Comment (natalieesk) [reply README.md:45]: Yes. The README is also for anyone contributing, whether that be other zfnd engineers or a third party.

Comment (conradoplg) [reply README.md:45]: These could be inside a `# Development Information` section or something similar

**Review by conradoplg — COMMENTED** [review]
This looks good, thanks! I think it just needs addressing Pili's comments.

File: README.md
Comment (natalieesk) [reply README.md:41]: I've put this in a "developer information" section as suggested.

File: README.md
Comment (natalieesk) [reply README.md:33]: Yeh I've moved it to the top under the main description. People will scroll down for more info as it's important context before continuing

File: README.md
Comment (natalieesk) [reply README.md:20]: Done

File: README.md
Comment (natalieesk) [reply README.md:31]: Done

File: README.md
Comment (conradoplg) [inline README.md:78]:
```suggestion
The dealer should use a secure broadcast channel to ensure each participant has a consistent view of this commitment and each participant must [verify the commitments](https://github.com/ZcashFoundation/frost/blob/4055cb9439df2814800c678c8da1760a0f86dc10/frost-core/src/frost/keys.rs#L297). 
```

The VSS verification must always be done, this was recently fixed in the spec.

File: README.md
Comment (conradoplg) [inline README.md:82]: The phrasing here is a bit confusing. The dealer can also be a signer.

I think we can remove the note, but here's a suggestion (assuming I got its intention)

```suggestion
NOTE: A signer is a participant that generates a signing share which is aggregated with the signing shares of other signers to obtain the final group signature.
```

**Review by conradoplg — CHANGES_REQUESTED** [review]
The fixes look good, but I noticed some other issues, sorry for not noticing them in the first pass.

File: README.md
Comment (natalieesk) [reply README.md:82]: Yeh that is confusing. I was trying to use the right terminology for a participant and a signer as described [here](https://frost.zfnd.org/dev/rfcs/0001-messages.html). Perhaps I can just leave that bit out though. I do think the distinction between participant and signer is a bit confusing and I was debating how to use it properly in the description.

File: README.md
Comment (natalieesk) [reply README.md:78]: Ah ok, thanks.

**Comment by natalieesk** [comment]
> The fixes look good, but I noticed some other issues, sorry for not noticing them in the first pass.

I added in more things based on the suggestions, so this wasn't there the first time round :-)

**Review by conradoplg — APPROVED** [review]

**Comment by conradoplg** [comment]
Looks good to me, waiting for @mpguerra 's approval

**Review by mpguerra — APPROVED** [review]

**Pull request merged**
